visualization/app/codeCharta/ui/screenshotButton/screenshotButton.component.spec.ts

```diff
@@ -1,95 +1,39 @@
-			threeSceneService,
-			threeCameraService,
-			threeRendererService,
-			storeService,
-			$rootScope
-		)
-	}
-
-	describe("makeScreenshot", () => {
-		it("should call loadScript", () => {
-			screenshotButtonController["loadScript"] = jest.fn()
-			screenshotButtonController.makeScreenshotToFile()
-			expect(screenshotButtonController["loadScript"]).toBeCalled()
-		})
-		it("should call makePNGFileName", () => {
-			screenshotButtonController["makePNGFileName"] = jest.fn()
-			screenshotButtonController.makeScreenshotToFile()
-			expect(screenshotButtonController["makePNGFileName"]).toBeCalled()
-		})
-
-		it("should call buildScreenShotCanvas", () => {
-			screenshotButtonController["buildScreenShotCanvas"] = jest.fn()
-			screenshotButtonController.makeScreenshotToFile()
-			expect(screenshotButtonController["buildScreenShotCanvas"]).toBeCalled()
+		TestBed.configureTestingModule({
+			imports: [ScreenshotButtonModule],
+			providers: [
+				{ provide: ThreeCameraServiceToken, useValue: {} },
+				{ provide: ThreeSceneServiceToken, useValue: {} },
+				{ provide: ThreeRendererServiceToken, useValue: {} }
+			]
 		})
 	})
 
-	describe("onScreenshotToClipboardEnabledChanged", () => {
-		it("should set screenshotToClipboardEnabled in viewModel", () => {
-			storeService.dispatch(setScreenshotToClipboardEnabled(true))
+	it("should copy to clipboard on click, when screenshot to clipboard is enabled", async () => {
+		Store.store.dispatch(setScreenshotToClipboardEnabled(true))
+		const { fixture } = await render(ScreenshotButtonComponent, { excludeComponentDeclaration: true })
+		fixture.componentInstance.makeScreenshotToClipBoard = jest.fn()
 
-			screenshotButtonController.onScreenshotToClipboardEnabledChanged(
-				storeService.getState().appSettings.screenshotToClipboardEnabled
-			)
-
-			expect(screenshotButtonController["_viewModel"].screenshotToClipboardEnabled).toBe(true)
-		})
+		userEvent.click(screen.getByTitle("Copy screenshot to clipboard (Ctrl+Alt+F), export it as a file by (Ctrl+Alt+S)"))
+		expect(fixture.componentInstance.makeScreenshotToClipBoard).toHaveBeenCalled()
 	})
 
-	describe("makeScreenshotToClipBoard", () => {
-		it("should call buildScreenShotCanvas", () => {
-			screenshotButtonController["buildScreenShotCanvas"] = jest.fn()
-			screenshotButtonController.makeScreenshotToFile()
-			expect(screenshotButtonController["buildScreenShotCanvas"]).toBeCalled()
-		})
+	it("should save to file on click, when screenshot to clipboard is not enabled", async () => {
+		Store.store.dispatch(setScreenshotToClipboardEnabled(false))
+		const { fixture } = await render(ScreenshotButtonComponent, { excludeComponentDeclaration: true })
+		fixture.componentInstance.makeScreenshotToFile = jest.fn()
+
+		userEvent.click(screen.getByTitle("Export screenshot as file (Ctrl+Alt+S), copy it to clipboard by (Ctrl+Alt+F)"))
+		expect(fixture.componentInstance.makeScreenshotToFile).toHaveBeenCalled()
```

Comment:
This test would pass if we would replace
`
makeScreenShotToFile = ( ) => { }`

Is there a way how we can make this test more "demanding" (=stricter assert part)?